memory-store-redis/deployment/src/test/java/io/quarkiverse/langchain4j/memorystore/redis/test/RedisChatMemoryStoreTest.java

```diff
@@ -0,0 +1,225 @@
+import static io.quarkiverse.langchain4j.memorystore.redis.test.MessageAssertUtils.assertMultipleRequestMessage;
+import static io.quarkiverse.langchain4j.memorystore.redis.test.MessageAssertUtils.assertSingleRequestMessage;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.stubbing.ServeEvent;
+import com.github.tomakehurst.wiremock.verification.LoggedRequest;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.service.MemoryId;
+import dev.langchain4j.service.UserMessage;
+import dev.langchain4j.store.memory.chat.ChatMemoryStore;
+import io.quarkiverse.langchain4j.ChatMemoryRemover;
+import io.quarkiverse.langchain4j.RegisterAiService;
+import io.quarkiverse.langchain4j.memorystore.RedisChatMemoryStore;
+import io.quarkus.redis.datasource.RedisDataSource;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class RedisChatMemoryStoreTest {
+
+    public static final int FIRST_MEMORY_ID = 1;
+    public static final int SECOND_MEMORY_ID = 2;
+    private static final int WIREMOCK_PORT = 8089;
+    private static final String API_KEY = "test";
+
+    @RegisterExtension
+    static final QuarkusUnitTest unitTest = new QuarkusUnitTest()
+            .setArchiveProducer(
+                    () -> ShrinkWrap.create(JavaArchive.class).addClasses(WiremockUtils.class, MessageAssertUtils.class))
+            .overrideRuntimeConfigKey("quarkus.langchain4j.openai.api-key", API_KEY)
+            .overrideRuntimeConfigKey("quarkus.langchain4j.openai.base-url", "http://localhost:" + WIREMOCK_PORT + "/v1");
+    private static final TypeReference<Map<String, Object>> MAP_TYPE_REF = new TypeReference<>() {
+    };
+
+    static WireMockServer wireMockServer;
+
+    static ObjectMapper mapper;
+
+    @BeforeAll
+    static void beforeAll() {
+        wireMockServer = new WireMockServer(options().port(WIREMOCK_PORT));
+        wireMockServer.start();
+
+        mapper = new ObjectMapper();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        wireMockServer.stop();
+    }
+
+    @BeforeEach
+    void setup() {
+        wireMockServer.resetAll();
+        wireMockServer.stubFor(WiremockUtils.defaultChatCompletionsStub(API_KEY));
+    }
```

Comment:
Can we please use the utilities that are now in `quarkus-langchain4j-testing-internal`? It should make this code unnecessary

Reply:
I'm trying this out right now but honestly I'm a bit lost

Reply:
Let me know if you need any help

Reply:
should be good now